third_party/move/tools/move-linter/tests/model_ast_lints/find_unnecessary_casts_test.move

```diff
@@ -0,0 +1,113 @@
+module 0xc0ffee::unnecessary_casts_test {
+
+    const U8_VALUE: u8 = 255;
+    const U64_VALUE: u64 = 1000;
+
+    fun helper_u64(x: u64): u64 { x }
+    fun helper_u8(x: u8): u8 { x }
+
+    // ========== POSITIVE TESTS (should warn) ==========
+
+    public fun test_cast_in_let_warn() {
+        let x: u64 = 42;
+        let y = (x as u64);
+    }
+
+    public fun test_cast_in_function_arg_warn() {
+        let x: u64 = 100;
+        let result = helper_u64(x as u64);
+    }
+
+    public fun test_cast_in_return_warn(): u64 {
+        let x: u64 = 42;
+        (x as u64)
+    }
+
+    public fun test_cast_in_expression_warn() {
+        let a: u32 = 10;
+        let b: u32 = 20;
+        let result = ((a + b) as u32);
+    }
+
+    public fun test_cast_const_warn() {
+        let x = (U8_VALUE as u8);
+    }
+
+    public fun test_cast_in_conditional_warn() {
+        let x: u64 = 5;
+        if ((x as u64) > 0) { };
+    }
+
+    public fun test_cast_in_assert_warn() {
+        let x: u16 = 100;
+        assert!((x as u16) > 0, 0);
+    }
+
+    public fun test_multiple_casts_warn() {
```

Comment:
Could we also add a test for nested casts, something like `((x as u64) as u64) `?